docs/modules/spring/pages/overview.adoc

```diff
@@ -1,7 +1,31 @@
 = Integrate with Spring
 
-You can integrate Hazelcast with Spring and this section explains the
-configuration of Hazelcast within Spring context.
+https://spring.io/[Spring] is a popular framework for writing enterprise-grade Java applications. https://spring.io/projects/spring-framework[Spring Framework] includes a range of libraries to reduce the need for boilerplate code and encourages a consistent structure for ease of maintenance. https://spring.io/projects/spring-boot[Spring Boot] further simplifies development by automatically configuring applications on your classpath.
 
-Supported Versions are Spring 4.3 and higher releases and the latest
-tested Spring version is 5.3.
+Hazelcast Platform is integrated with the Spring Framework and Spring Boot, allowing you to easily incorporate Hazelcast's fast data cache capabilities into your Spring applications.
+
+== Why use Hazelcast with Spring?
+
+You can use Hazelcast to enhance your Spring applications with caching and session replication.
+
+**Cache** - You can use Hazelcast as a Spring cache manager or a JCache-compliant cache, as described in the https://docs.spring.io/spring-boot/reference/io/caching.html#io.caching.provider.hazelcast[Spring Boot documentation].
+
+**Hibernate second-level cache** - You can use Hazelcast as a second-level cache in the https://hibernate.org/[Hibernate] framework.
```

Comment:
I just realised that this use case is separate from Spring integration and it would be _more_ correct to put it under Hibernate integration chapter.

We can mention it though here:

```suggestion
**Hibernate second-level cache** - You can use Hazelcast as a second-level cache in your https://hibernate.org/[Hibernate] `LocalSessionFactoryBean` configuration.
```

as it's commonly used with Spring